File: 0229-majority-element-ii/0229-majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        vector<int> ans;
        int n = nums.size();
        int f = n/3;

        unordered_map<int,int> mpp;

        for(int it: nums){
            mpp[it]++;
        }

        for(auto it: mpp){
            if(it.second > f){
                ans.push_back(it.first);
            }
        }

        return ans;
    }
};

File: 0229-majority-element-ii/0229-majority-element-ii.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        int n=nums.length;
        int m=n/3;

        List<Integer> lst = new ArrayList<>();
        HashMap<Integer,Integer> mpp = new HashMap<>();

        for(int i: nums){
            mpp.put(i,mpp.getOrDefault(i,0)+1);
        }

        for(Map.Entry<Integer,Integer> it: mpp.entrySet()){
            if(it.getValue()>m){
                lst.add(it.getKey());
            }
        }

        return lst;
    }
}